lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3294,18 +3294,52 @@ private function performCallbackOnCachedPersister(callable $callback)
         }
     }
 
+    private function dispatchFlushEntityListener($event)
+    {
+        $subscribedSystems = [];
+        foreach ($this->entityChangeSets as $oid => $entityChangeSet) {
+            foreach ($this->identityMap as $className => $entities) {
+                $hash = (isset($this->entityIdentifiers[$oid], $this->entityIdentifiers[$oid]['id'])) ? $this->entityIdentifiers[$oid]['id'] : null;
+                if (!$hash) {
+                    continue;
+                }
+                $hash = (string) $hash;
```

Comment:
I was having problems with the tests trying to pass in Objects as IDs, so I implemented the Type Cast to pass said tests.